### DISCUSSION TITLE
ResponsiveLine, Typescript

### DISCUSSION THREAD

**Discussion by ricktoews** [plouc/nivo#2004] [Q&A]
Attempting to use ResponsiveLine with TypeScript, getting "No overload matches this call." error.
It looks as though this is an example of a long-known issue with TypeScript definitions in nivo.

I saw that there's a line-typescript branch: https://github.com/plouc/nivo/tree/line-typescript/packages

Can ResponsiveLine be used with TypeScript? If so, what needs to be modified? Would one need to go into node_modules/@nivo/line and add a types directory, for instance?

**Comment by ricktoews** [comment]
Follow-up: I found the suggestion to add `declare module '@nivo/line'` here: https://github.com/plouc/nivo/issues/197#issuecomment-391347241, and I put it in what looked like a global DTS file in my project (react-app-env.d.ts). 

That appears to have fixed the issue.

If there's a more orthodox way to deal with the situation, I'd be interested in learning it.

**Comment by hyjoong** [comment]
I'm having the same issues.
An error occurred when I copied and pasted the code on the official website.

 

orient values were removed from attribute `axisBottom` and attribute `axisLeft`

```diff
axisBottom={{
-orient: 'left',
```
